buildSrc/src/main/kotlin/conventions/detekt.gradle.kts

```diff
@@ -0,0 +1,44 @@
+/*
+ * Copyright 2008-present MongoDB, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package conventions
+
+import io.gitlab.arturbosch.detekt.Detekt
+
+// Static code analysis for Kotlin
+// https://plugins.gradle.org/plugin/io.gitlab.arturbosch.detekt
+plugins { id("io.gitlab.arturbosch.detekt") }
+
+detekt {
+    allRules = true // fail build on any finding
+    buildUponDefaultConfig = true // preconfigure defaults
+    config = rootProject.files("config/detekt/detekt.yml") // point to your custom config defining rules to run,
```

Comment:
Any reason while this is `.files` but the one below is `.file`?  If not, let's make them consistent.

Reply:
Yes, its because config requires a `ConfigurableFileCollection`.